### PR TITLE
feat(wallet setup): mnemonic input word suggestion redesign

### PR DESCRIPTION
**Pull request by banklesss**
Almost the entire state has been moved up to the parent component. 
Removed useEffects to make state changes synchronous

related to [YOMO-1364](https://emurgo.atlassian.net/browse/YOMO-1364)

[YOMO-1364]: https://emurgo.atlassian.net/browse/YOMO-1364?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jorbuedo — APPROVED** [review]

File: apps/wallet-mobile/src/features/SetupWallet/common/MnemonicInput/MnemonicInput.tsx [hunk 1]
```diff
@@ -208,95 +224,56 @@ const MnemonicWordInput = React.forwardRef<RNTextInput, MnemonicWordInputProps>(
...
-        }
-        visible={menuEnabled && word.length > 0 && !isEmptyString(word)}
-        onDismiss={() => {
-          setMenuEnabled(false)
-          setWord('')
+      <TextInput
+        ref={inputRef}
+        value={word}
+        onFocus={(e) => {
+          // selectTextOnFocus is buggy on ios
```
Comment (stackchain) [inline apps/wallet-mobile/src/features/SetupWallet/common/MnemonicInput/MnemonicInput.tsx:243]: it was fixed, we need to bump .73

Comment (banklesss) [reply apps/wallet-mobile/src/features/SetupWallet/common/MnemonicInput/MnemonicInput.tsx:243]: i tested and still not working on ios : |

Comment (banklesss) [reply apps/wallet-mobile/src/features/SetupWallet/common/MnemonicInput/MnemonicInput.tsx:243]: i will check it again when i work on the bump 👍🏻

Comment (stackchain) [reply apps/wallet-mobile/src/features/SetupWallet/common/MnemonicInput/MnemonicInput.tsx:243]: on .73?

**Pull request merged**